pycheribuild/projects/cross/apache.py

```diff
@@ -129,3 +129,36 @@ def configure(self, **kwargs):
                          cwd=self.build_dir / "server")
             self.run_cmd(str(self.host_CC), "gen_test_char.lo", "-o",
                          "gen_test_char", cwd=self.build_dir / "server")
+
+class BuildSSLProc(CrossCompileCMakeProject):
+    target = "sslproc"
+
+    repository = GitRepository("https://github.com/CTSRD-CHERI/sslproc.git")
+
+    native_install_dir = DefaultInstallDir.IN_BUILD_DIRECTORY
+    cross_install_dir = DefaultInstallDir.ROOTFS_LOCALBASE
+
+    @classmethod
+    def setup_config_options(cls, **kwargs):
+        super().setup_config_options(**kwargs)
+        cls.build_tests = cls.add_bool_option("build-tests", help="Build tests",
+                                              show_help=True)
+
+    def setup(self):
+        super().setup()
+        if self.build_tests:
+            self.configure_args.append("-DBUILD_TESTS=ON")
+
+class BuildSSLProcApache(BuildApache):
+    target = "apache-sslproc"
+    default_directory_basename = "apache"
+    build_dir_suffix = "-sslproc"
+
+    # XXX: Can this extend the list from BuildApache?
```

Comment:
Not sure if there's a better way to do this.  I'd really like to be able to say something like:

```
    dependencies.append("sslproc")
```

assuming the super() constructor had run first, but maybe that doesn't work?  Would I need to do something like:

```
    dependencies = BuildApache.dependencies + ["sslproc"]
```

Reply:
We raced here. The latter works. The former doesn't at that scope, but we do let dependencies be a method instead, which would mean writing it as:
```
    @classmethod
    def dependencies(cls, config):
        return super().dependencies + ["sslproc"]
```
(yay dynamic typing?..)

Reply:
Umm, I'm going to go with your suggestion (option 2 above).  The other feels a bit too gross. :)